### PR TITLE
[CFM-1411] event from overview page

### PR DESCRIPTION
**Pull request by nirgn975**
https://issuetracker.amplexor.com/jira/browse/CFM-1411

### CODE REVIEW

File: project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_events/c4m_features_og_events.module [hunk 1]
```diff
@@ -121,6 +121,33 @@ function c4m_features_og_events_views_pre_render(&$view) {
...
+  if (!in_array($variables['name'], Array('c4m_overview_og_events_landing', 'c4m_overview_og_events'))) {
+    return;
+  }
+
+  if (!in_array($variables['display_id'], Array('page', 'page_1'))) {
+    return;
+  }
+
+  $icon = '<i class="fa fa-calendar-plus-o" aria-hidden="true"></i>';
+  $label = 'Add an event';
```
Comment (nirgn975) [inline project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_events/c4m_features_og_events.module]: t()

File: project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_events/c4m_features_og_events.module [hunk 2]
```diff
@@ -121,6 +121,33 @@ function c4m_features_og_events_views_pre_render(&$view) {
...
+
+  // Check if the user has permission to add an event.
+  if (og_user_access(og_context()['group_type'], og_context()['gid'], 'create event content')) {
+    $variables['header'] = str_replace(strtoupper($label), $icon . ' ' . $label, $variables['header']);
+  }
+  else {
```
Comment (nirgn975) [inline project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_events/c4m_features_og_events.module]: move above

File: project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_events/c4m_features_og_events.module [hunk 3]
```diff
@@ -121,6 +121,33 @@ function c4m_features_og_events_views_pre_render(&$view) {
...
+    return;
+  }
+
+  if (!in_array($variables['display_id'], Array('page', 'page_1'))) {
+    return;
+  }
+
+  $icon = '<i class="fa fa-calendar-plus-o" aria-hidden="true"></i>';
+  $label = 'Add an event';
+
```
Comment (nirgn975) [inline project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_events/c4m_features_og_events.module]: `$og_context = og_context();`

**Comment by nirgn975** [comment]
@ordavidil 
Two of tests fails because anonymous user should have access to create an event (see test `Check upcoming events overview as an anonymous user` and `Check past events overview as an anonymous user` in `group_events.feature`).

And we check `create event content`.

**Comment by ordavidil** [comment]
@nirgn975 
Please solve ___Scenario: Check events landing page as group owner___ first.

**Comment by nirgn975** [comment]
@ordavidil Solve the **_Scenario: Check events landing page as group owner_**.

Now just the **_Scenario: Check upcoming events overview as an anonymous user_** and **_Scenario: Check past events overview as an anonymous user_** tests are fail. And I saw you asked Kevin about anonymous users and he said no. But it looks like the test check if anonymous **has** permission to create an event. So, what should I do here?

**Comment by ordavidil** [comment]
@nirgn975 
please ask @janlemmens for clarification on JIRA.

File: project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_events/c4m_features_og_events.module [hunk 4]
```diff
@@ -122,6 +122,36 @@ function c4m_features_og_events_views_pre_render(&$view) {
...
+  if (!in_array($variables['display_id'], array('page', 'page_1'))) {
+    // Not a "page" display.
+    return;
+  }
+
+  $og_context = og_context();
+
+  if (!og_user_access($og_context['group_type'], $og_context['gid'], 'create event content')) {
+    // User doesn't have permission to create a new event.
+    $variables['header'] = '';
```
Comment (nirgn975) [inline project/profiles/capacity4more/modules/c4m/features_og/c4m_features_og_events/c4m_features_og_events.module]: @ordavidil This is our problem.

**Comment by nirgn975** [comment]
@ordavidil Ready for review, finally!

**Comment by ordavidil** [comment]
@nirgn975 
What about a nice screenshot ?

**Comment by nirgn975** [comment]
@ordavidil You are right sir.
capacity4more
upcoming_events___capacity4more

**incognito**
upcoming_events___capacity4more

**Comment by ordavidil** [comment]
@nirgn975 
Where is the sort button on the first image ?

**Comment by guytepper** [comment]
@nirgn975, In order to avoid code repetition I moved the button modification function to a new module, check #1050.

Also there is a [fix](https://github.com/capacity4dev/capacity4more/pull/1050/files#diff-9dd6600b8e7034a4609069f44cb1e1ecR56) for an issue when the button showing up when the user doesn't have permission

**Comment by ordavidil** [comment]
@guytepper 
since you checked out from this branch, please fix that on your branch ...

@nirgn975 
Leave this branch as is to avoid conflicts ...

**Comment by guytepper** [comment]
@ordavidil what do you mean? The fix is on my branch.

**Comment by guytepper** [comment]
Closing in favor of #1081 .

**Pull request closed without merging**